src/dis6/VariableDatum.cpp

```diff
@@ -30,15 +27,15 @@ void VariableDatum::setVariableDatumID(unsigned int pX)
 
 unsigned int VariableDatum::getVariableDatumLength() const
 {
-    return _variableDatumLength;
+    return _variableDatums.size();
```

Comment:
I think this needs to by multiplied by 8. Assuming it is the Variable Datum Length from the Standard.
See the previous line 59, store the `length * 8` into `_variableDatumLength`.